src/main/java/com/demoipm/service/impl/UserAppServiceImpl.java

```diff
@@ -114,6 +132,159 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
                 userApp.getPassword(), grantList);
 
         return userDetails;
-	}
+    }
+
+    @Override
+    public UserListPageResponseDto readByCondition(String searchWord, int pageNo, int entriesNo) {
+        LOGGER.info("Start readByCondition with searchWord {}, pageNo {}, entriesNo {}", searchWord, pageNo, entriesNo);
+        UserListPageResponseDto responseDto = new UserListPageResponseDto();
+        try {
+            // Get data from database with pagination and search word
+            Pageable pageable = PageRequest.of(pageNo - 1, entriesNo, Sort.Direction.ASC, "username");
+            Page<UserApp> userPage = userAppDao.findListByCondition("%" + searchWord + "%", pageable);
+
+            // Prepare response dto
+            responseDto.setSearchWord(searchWord);
+            responseDto.setCurrentPage(pageNo);
+            responseDto.setEntriesNo(entriesNo);
+            responseDto.setTotalPage(userPage.getTotalPages());
+            responseDto.setTotalEntries(userPage.getTotalElements());
+            responseDto.setUserList(userPage.getContent().stream().map(userApp -> {
```

Comment:
separate this return function before setting to another object

Reply:
Fixed. Please review